### ISSUE TITLE
Unsupported type: Mediaplayer (Sonos)

### ISSUE THREAD

**Issue by gerard33** [t0mg/reacticz#38]
@t0mg, is it possible to add support for the Mediaplayer switch? Yes another switch ;)

I use that one for a.o. my Sonos. It's shows what's playing on the Sonos and you can switch it on/off (the remote in the screenshot is not used).

![mediaplayer](https://cloud.githubusercontent.com/assets/11230573/22309469/abc6c182-e34b-11e6-8afe-744e059f1b79.png)

Updates are being send with MQTT for this type of switch.

`{"Battery":255,"RSSI":12,"dtype":"Light/Switch","id":"00014236","idx":486,"name":"Sonos kantoor - playing","nvalue":1,"stype":"Switch","svalue1":"The Weeknd - I Feel It Coming","switchType":"Media Player","unit":1}`

```
{
   "ActTime" : 1485378918,
   "ServerTime" : "2017-01-25 22:15:18",
   "Sunrise" : "08:28",
   "Sunset" : "17:07",
   "result" : [
      {
         "AddjMulti" : 1.0,
         "AddjMulti2" : 1.0,
         "AddjValue" : 0.0,
         "AddjValue2" : 0.0,
         "BatteryLevel" : 255,
         "CustomImage" : 123,
         "Data" : "The Weeknd - I Feel It Coming",
         "Description" : "",
         "Favorite" : 1,
         "HardwareID" : 3,
         "HardwareName" : "Dummy",
         "HardwareType" : "Dummy (Does nothing, use for virtual switches only)",
         "HardwareTypeVal" : 15,
         "HaveDimmer" : true,
         "HaveGroupCmd" : true,
         "HaveTimeout" : false,
         "ID" : "00014236",
         "Image" : "Sonos1",
         "IsSubDevice" : false,
         "LastUpdate" : "2017-01-25 22:15:09",
         "Level" : 0,
         "LevelInt" : 0,
         "MaxDimLevel" : 100,
         "Name" : "Sonos kantoor - playing",
         "Notifications" : "false",
         "PlanID" : "0",
         "PlanIDs" : [ 0 ],
         "Protected" : false,
         "ShowNotifications" : true,
         "SignalLevel" : "-",
         "Status" : "On",
         "StrParam1" : "",
         "StrParam2" : "",
         "SubType" : "Switch",
         "SwitchType" : "Media Player",
         "SwitchTypeVal" : 17,
         "Timers" : "false",
         "Type" : "Light/Switch",
         "TypeImg" : "Media",
         "Unit" : 1,
         "Used" : 1,
         "UsedByCamera" : false,
         "XOffset" : "0",
         "YOffset" : "0",
         "idx" : "486"
      }
   ],
   "status" : "OK",
   "title" : "Devices"
}
```

**Comment by t0mg** [comment]
Similar to issue #25 but not exactly the same. In the MQTT message,  switchType is "Media Player" in both cases but dtype  is "Light/Switch" instead of "Lighting 2", stype is "Switch" instead of "AC".

Could you do the same as I did in #25, ie listing the commands of the remote widget ? Just on/off and play pause for now as I am still not sure that I want a full size remote in reacticz.

Thanks

**Comment by gerard33** [comment]
For the Sonos part on and off would be enough. That is what I am using the switch for right now.
The remote of that switch is not working.
But the special part of this switch is that it shows the info on the song that is being played.
Would be great if you could show that info in Reacticz as well.

I also have a Mediaplayer switch for my Kodi. That one is indeed "Lighting 2" and "AC".
It also shows the title of what is playing on Kodi. The remote of that switch is working, but I guess on/off and play/pause would indeed be enough.

Let me know if you need some more info.

**Comment by gerard33** [comment]
Made pull request #41. Let me know what you think of it.

**Comment by gerard33** [comment]
Merged in #41.
